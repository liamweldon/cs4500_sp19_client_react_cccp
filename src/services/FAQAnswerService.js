import {API_ROOT} from '../api-config';

export default class FAQAnswerService {
    static instance = null;
    static getInstance() {
        if (FAQAnswerService.instance === null) {
            FAQAnswerService.instance = new FAQAnswerService();
        }
        return this.instance;
    }
    findFAAById = (id) => fetch(`${API_ROOT}/api/faas/${id}`).then((response) => response.json());
    findAllFAAs = () => fetch(`${API_ROOT}/api/faas`).then((response) => response.json());
    addFAA = (newAnswer) =>
        fetch(`${API_ROOT}/api/faas/`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(newAnswer)
        }).then((response) => response.json());

    editFAA = (id, newAnswer) =>
        fetch(`${API_ROOT}/api/faas/${id}`, {
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(newAnswer)
        }).then((response) => response.json());
}