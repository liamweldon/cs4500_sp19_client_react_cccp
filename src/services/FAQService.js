import {API_ROOT} from '../api-config';

export default class FAQService {
  static instance = null;
  static getInstance() {
    if (FAQService.instance === null) {
      FAQService.instance = new FAQService();
    }
    return this.instance;
  }

  addFAQ = (newQuestion) =>
    fetch(`${API_ROOT}/api/faqs/`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(newQuestion)
    }).then((response) => response.json());

  editFAQ = (id, newQuestion) =>
    fetch(`${API_ROOT}/api/faqs/${id}`, {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(newQuestion)
    }).then((response) => response.json());
  findFAQById = (id) => fetch(`${API_ROOT}/api/faqs/${id}`).then((response) => response.json());
  deleteFAQ = (id) => fetch(`${API_ROOT}/api/faqs/${id}`, {method: 'DELETE'}).then((response) => response.json());
  findAllFAQs = () => fetch(`${API_ROOT}/api/faqs`).then((response) => response.json());
  linkFAAtoFAQ = (faaID, faqID) => fetch(`${API_ROOT}/api/faqs/${faqID}/faas/${faaID}`).then(response => response.json());
}
