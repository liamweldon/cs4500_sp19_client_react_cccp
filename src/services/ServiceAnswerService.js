export default class ServiceAnswerService {
    static instance = null;
    static getInstance() {
        if(ServiceAnswerService.instance === null) {
            ServiceAnswerService.instance = new ServiceAnswerService()
        }
        return this.instance
    }
    findServiceAnswerById = id =>
    fetch(`http://localhost:8080/api/service-answers/${id}`)
    .then(response => response.json())
    findAllServiceAnswers = () =>
    fetch("http://localhost:8080/api/service-answers")
    .then(response => response.json())
}