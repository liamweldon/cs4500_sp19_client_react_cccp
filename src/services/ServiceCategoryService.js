export default class ServiceCategoryService {
    static instance = null;
    static getInstance() {
        if(ServiceCategoryService.instance === null) {
            ServiceCategoryService.instance = new ServiceCategoryService()
        }
        return this.instance
    }
    findServiceCategoryById = categoryId =>
        fetch(`http://localhost:8080/api/service-categories/${categoryId}`)
            .then(response => response.json())
    findAllServiceCategories = () =>
        fetch("http://localhost:8080/api/service-categories")
            .then(response => response.json())
}
