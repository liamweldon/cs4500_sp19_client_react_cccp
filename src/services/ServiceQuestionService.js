import {API_ROOT} from '../api-config';

export default class ServiceQuestionService {
    static instance = null;
    static getInstance() {
        if (ServiceQuestionService.instance === null) {
            ServiceQuestionService.instance = new ServiceQuestionService();
        }
        return this.instance;
    }
    findServiceQuestionById = (id) =>
        fetch(`${API_ROOT}/api/service-questions/${id}`).then((response) => response.json());

    findAllServiceQuestions = () => fetch(`${API_ROOT}/api/service-questions`).then((response) => response.json());

    addServiceQuestion = (newServiceQuestion) =>
    fetch(`${API_ROOT}/api/service-questions/`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(newServiceQuestion)
    }).then((response) => response.json());
}
