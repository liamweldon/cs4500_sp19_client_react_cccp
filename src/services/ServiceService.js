import { API_ROOT } from "../api-config";

export default class ServiceService {
  static instance = null;

  static getInstance() {
    if (ServiceService.instance === null) {
      ServiceService.instance = new ServiceService();
    }
    return this.instance;
  }

  findServiceById = serviceId =>
    fetch(`${API_ROOT}/api/services/${serviceId}/`).then(response =>
      response.json());

  findAllServices = () =>
    fetch(`${API_ROOT}/api/services`).then(response => response.json());

  createService = service => {
    delete service.id;
    return fetch(`${API_ROOT}/api/services`,
    {
      method: 'post',
      body: JSON.stringify(service),
      headers: {'content-type': 'application/json'}
    })
    .then(response => response.json())
  };

  updateService = service => {
    return fetch(`${API_ROOT}/api/services/${service.id}`,
    {
      method: 'put',
      body: JSON.stringify(service),
      headers: {'content-type': 'application/json'}
    })
    .then(response => response.json())
  };

  deleteService = id => {
    return fetch(`${API_ROOT}/api/services/${id}`,
    {
      method: 'delete'
    })
  };

}
