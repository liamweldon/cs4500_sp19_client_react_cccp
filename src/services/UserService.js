import {API_ROOT} from '../api-config';

export default class UserService {
    static instance = null;

    static getInstance() {
        if (UserService.instance === null) {
            UserService.instance = new UserService();
        }
        return this.instance;
    }

    findUserById = (userId) => fetch(`${API_ROOT}/api/users/${userId}`).then((response) => response.json());
    findAllUsers = () => fetch(`${API_ROOT}/api/users`).then((response) => response.json());
    // TODO update urls
    createUser = user =>
        fetch(`${API_ROOT}/api/users`,
            {
                method: 'POST',
                body: JSON.stringify(user),
                headers: {'content-type': 'application/json'}
            })
            .then(response => response.json())

    updateUser = user =>
        fetch(`${API_ROOT}/api/users/${user.id}`,
            {
                method: 'PUT',
                body: JSON.stringify(user),
                headers: {'content-type': 'application/json'}
            })
            .then(response => response.json())

    deleteUser = user =>
        fetch(`${API_ROOT}/api/users/${user.id}`,
            {
                method: 'DELETE',
            })
}
