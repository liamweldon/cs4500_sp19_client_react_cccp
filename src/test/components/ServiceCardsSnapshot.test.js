import React from "react";
import TestRenderer from "react-test-renderer";

import ServiceCards from "../../components/ServiceNavigator/ServiceCards";
import ServiceService from "../../services/ServiceService";

import services from "../mockData/services.mock.json";
import mockServiceService from "../services/mocks/MockServiceService";

const serviceService = ServiceService.getInstance();
beforeEach(() => {
  global.fetch = mockServiceService(services);
});

// Provide a new service to insert
const test_new_service = {
  id: 555,
  serviceName: "Car Wash",
  serviceDescription: "Clean your car!"
};


// Snapshot testing code
test("Creation renders properly in DOM", () => {
  // Create snapshot of initial services
  return serviceService
    .findAllServices()
    .then(services => {
      let component = TestRenderer.create(
        <ServiceCards services={services} />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      // Create a new service via our mocked ServiceService. In the normal case,
      // the ServiceService would modify the database, and the ServiceContainer
      // would then update its state with the new data and re-call the Services
      // funcional component. Here, we simulate that process by updating our
      // mock data via the mock ServiceService and then recreating our
      // functional component with the modified mock data.
      serviceService
        .createService(test_new_service)
        .then(service => {
          component = TestRenderer.create(
            <ServiceCards services={services} />
          );
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot();
        })
        .catch(function(error) {});
    })
    .catch(function(error) {});
});
