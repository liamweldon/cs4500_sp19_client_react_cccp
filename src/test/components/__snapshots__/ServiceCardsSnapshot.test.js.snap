// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creation renders properly in DOM 1`] = `
<div
  className="row"
>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          House Cleaners
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Interior Painters
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Math Tutoring
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Test1
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Test2
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Test3
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`Creation renders properly in DOM 2`] = `
<div
  className="row"
>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          House Cleaners
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Interior Painters
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Math Tutoring
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Test1
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Test2
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Test3
        </a>
      </p>
    </div>
  </div>
  <div
    className="card no-border col-6"
  >
    <img
      className="card-img-top"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        <a
          href="/provider"
        >
           
          Car Wash
        </a>
      </p>
    </div>
  </div>
</div>
`;
